src/Promitor.Agents.Scraper/Validation/MetricDefinitions/ResourceTypes/EventHubsMetricValidator.cs

```diff
@@ -17,9 +17,8 @@ public IEnumerable<string> Validate(MetricDefinition metricDefinition)
             Guard.NotNull(metricDefinition, nameof(metricDefinition));
 
             var errorMessages = new List<string>();
-
-            var configuredDimension = metricDefinition.AzureMetricConfiguration?.Dimension?.Name;
-            var isEntityNameDimensionConfigured = string.IsNullOrWhiteSpace(configuredDimension) == false && configuredDimension.Equals(EntityNameDimension, StringComparison.InvariantCultureIgnoreCase);
+            
+            var isEntityNameDimensionConfigured = metricDefinition.AzureMetricConfiguration?.Dimensions?.Where(dimension => dimension.Name.Equals(EntityNameDimension, StringComparison.InvariantCultureIgnoreCase)).Any() ?? false;
```

Comment:
Given the duplication, I think this would be a good extension method to add
```suggestion
            var isEntityNameDimensionConfigured = metricDefinition.AzureMetricConfiguration?.Dimensions?.HasDimension(EntityNameDimension);
```

Reply:
Yeah, I like that idea! Added in new commits.